HerPublicWebsite.BusinessLogic/Services/RegularJobs/ReferralFollowUpNotificationService.cs

```diff
@@ -0,0 +1,50 @@
+﻿using HerPortal.BusinessLogic;
+using HerPublicWebsite.BusinessLogic.ExternalServices.EmailSending;
+using HerPublicWebsite.BusinessLogic.Services.ReferralFollowUps;
+using Microsoft.Extensions.Options;
+
+namespace HerPublicWebsite.BusinessLogic.Services.RegularJobs;
+
+public interface IReferralFollowUpNotificationService
+{
+    public Task SendReferralFollowUpNotifications();
+}
+
+public class ReferralFollowUpNotificationService : IReferralFollowUpNotificationService
+{
+    private readonly ReferralFollowUpNotificationServiceConfiguration config;
+    private readonly IDataAccessProvider dataProvider;
+    private readonly CsvFileCreator.CsvFileCreator csvFileCreator;
+    private readonly IWorkingDayHelperService workingDayHelperService;
+    private readonly IReferralFollowUpService referralFollowUpManager;
+    private readonly IEmailSender emailSender;
+
+    public ReferralFollowUpNotificationService(
+        IOptions<ReferralFollowUpNotificationServiceConfiguration> options,
+        IEmailSender emailSender,
+        IDataAccessProvider dataProvider,
+        CsvFileCreator.CsvFileCreator csvFileCreator, 
+        IWorkingDayHelperService workingDayHelperService,
+        IReferralFollowUpService referralFollowUpManager
+        )
+    {
+        config = options.Value;
+        this.emailSender = emailSender;
+        this.dataProvider = dataProvider;
+        this.csvFileCreator = csvFileCreator;
+        this.workingDayHelperService = workingDayHelperService;
+        this.referralFollowUpManager = referralFollowUpManager;
+    }
+
+    public async Task SendReferralFollowUpNotifications()
+    {
+        var endDate = await workingDayHelperService.AddWorkingDaysToDateTime(DateTime.Today, -10);
+        var newReferrals = await dataProvider.GetReferralRequestsWithNoFollowUpBeforeDate(endDate);
+        foreach (var newReferral in newReferrals) {
+            var referralRequestFollowUp = await referralFollowUpManager.CreateReferralRequestFollowUp(newReferral);
+            emailSender.SendFollowUpEmail(newReferral, config.AppBaseUrl + "referral-follow-up?token=" + referralRequestFollowUp.Token);
```

Comment:
Not sure if there's a way we should be doing this without a hardcoded route?

Reply:
Hmm - there is if you have an httpcontext, but you won't have access to that here. It'd be nicer to construct this URL using proper methods (e.g. with escaping) but this is fine - we know the token won't contain any special characters.

Reply:
I've done a bit more to improve this. But let me know if you want me to change anything else.